library/core/src/escape.rs

```diff
@@ -149,77 +151,197 @@ const fn escape_unicode<const N: usize>(c: char) -> ([ascii::Char; N], Range<u8>
     (output, (start as u8)..(N as u8))
 }
 
+#[derive(Clone, Copy)]
+union MaybeEscapedCharacter<const N: usize> {
+    pub escaped: [ascii::Char; N],
+    pub unescaped: char,
```

Comment:
Seems to me like the variants should be called something closer to `escape_seq`/`escape` and `literal`/`lit`.